core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -113,7 +113,7 @@ protected TransportReplicationAction(Settings settings, String actionName, Trans
         transportService.registerRequestHandler(actionName, request, ThreadPool.Names.SAME, new OperationTransportHandler());
         transportService.registerRequestHandler(transportPrimaryAction, request, executor, new PrimaryOperationTransportHandler());
         // we must never reject on because of thread pool capacity on replicas
-        transportService.registerRequestHandler(transportReplicaAction, replicaRequest, executor, true,
+        transportService.registerRequestHandler(transportReplicaAction, replicaRequest, executor, true, true,
```

Comment:
@danielmitterdorfer One thing that we do with `TransportReplicationAction` is that we ensure that if a replication action is successful on a primary, it will *not* be subjected to queue length limits on replicas (you don't want request on replicas to be subjected to `EsRejectedExecutionException`). That's what the first `true` parameter corresponding to `forceExecution` here is doing. But it seems like we are subjecting the replication requests to the size limits and I think that that might be wrong, or at least inconsistent with how we manage the queues for replication requests. Can you clarify?